app/code/Magento/Catalog/Pricing/Price/CustomOptionCatalogRulePriceCalculator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Catalog\Pricing\Price;
+
+use Magento\Catalog\Model\Product\Option\Value as ProductOptionValue;
+use Magento\Catalog\Model\Product\PriceModifierInterface;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+
+/**
+ * Calculates prices of custom options of the product with catalog rules applied.
+ */
+class CustomOptionCatalogRulePriceCalculator
+{
+    /**
+     * @var PriceCurrencyInterface
+     */
+    private $priceCurrency;
+
+    /**
+     * @var PriceModifierInterface
+     */
+    private $priceModifier;
+
+    /**
+     * @param PriceCurrencyInterface $priceCurrency
+     * @param PriceModifierInterface $priceModifier
+     */
+    public function __construct(
+        PriceCurrencyInterface $priceCurrency,
+        PriceModifierInterface $priceModifier
+    ) {
+        $this->priceModifier = $priceModifier;
+        $this->priceCurrency = $priceCurrency;
+    }
+
+    /**
+     * Calculates prices of custom options of the product with catalog rules applied.
+     *
+     * @param ProductOptionValue $optionValue
+     * @return float
+     */
+    public function execute(
```

Comment:
The `execute()` method underscores the functional usage of a service class, that's a correct approach. 

For this reason, a better name for the class would be one that suggests the command that is executed, like `CalculateCustomOptionCatalogRule`.

Reply:
@aleron75 done.